---
output: github_document
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%",
  tidy.opts = list(width.cutoff = 80),
  tidy = TRUE
)
```

# COV19 MSA

A multiple-sequence alignment tool for SARS-CoV-2 viral genomes that removes the following homoplasic sites:

**187, 1059, 2094, 3037, 3130, 6990, 8022, 10323, 10741, 11074, 13408, 14786, 19684, 20148, 21137, 24034, 24378, 25563, 26144, 26461, 26681, 28077, 28826, 28854, 29700, 4050, 13402, 11083, 15324, 21575**

## Overview

The script options are the following (**ensure input_fasta is in same directory as cov19_msa**):
```
Rscript --verbose msa.R [input_fasta] [iterations] [refinements] [cores] 
```
* iterations: Number of iterations for multiple sequence alignment algorithm - 0 recommended unless sequences are particularly low quality (then 5-6). [0-n]
* refinements: Refinement steps for the multiple sequence alignment algorithm - 0 recommended unless sequences are particularly low quality (then 2-3). [0-n]
* cores: number of cores to utilize during processing [1-n]

E.g:
```
Rscript --verbose msa.R example.fasta 0 0 4
```

Output is saved to **out** folder under same name as input file

## Installation

Clone repository:
```
git clone https://github.com/hsmaan/cov19_msa.git
```

Set up and activate conda environment:
```
conda env create --name msa_env --file msa.yaml
conda activate msa_env
```

Example run:
```
Rscript --verbose msa.R example.fasta 0 0 4
```
